xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/Block.java

```diff
@@ -280,6 +281,62 @@ public interface Block extends Cloneable
      */
     void setParameters(Map<String, String> parameters);
 
+    /**
+     * @return all attributes
+     * @since 14.10.5
+     * @since 15.1RC1
+     */
+    @Unstable
+    default Map<String, Object> getAttributes()
+    {
+        return Map.of();
+    }
+
+    /**
+     * An attribute is a generic key/value pair which can be used to add internal data to a block.
+     * <p>
+     * These attribute shouldn't be serialized by any renderer, they are meant to be used internally, e.g., in an XDOM
+     * transformation or a macro execution to store data related to the block.
+     *
+     * @param name the name of the attribute to return
+     * @return the attribute or null if the attribute doesn't exist
+     * @since 14.10.5
+     * @since 15.1RC1
+     */
+    @Unstable
+    default Object getAttribute(String name)
+    {
+        return null;
+    }
+
+    /**
+     * Set an attribute on the current block. See {@link #getAttribute(String)} for more details.
+     *
+     * @param name the attribute's name
+     * @param value the attribute's value
+     * @since 14.10.5
+     * @since 15.1RC1
+     */
+    @Unstable
+    default void setAttribute(String name, Object value)
+    {
+        throw new UnsupportedOperationException();
```

Comment:
There is not much point in implementing this method to indicate that it's not implemented. Would be much more clear to have it listed as breakage.

Reply:
Okay, I've opened a [vote](https://forum.xwiki.org/t/api-breakage-on-block-to-add-attributes/11830) for the API breakage.